receiver/prometheusreceiver/internal/transaction.go

```diff
@@ -207,7 +211,10 @@ func (t *transaction) getMetrics(resource pcommon.Resource) (pmetric.Metrics, er
 	md := pmetric.NewMetrics()
 	rms := md.ResourceMetrics().AppendEmpty()
 	resource.CopyTo(rms.Resource())
-	metrics := rms.ScopeMetrics().AppendEmpty().Metrics()
+	ils := rms.ScopeMetrics().AppendEmpty()
+	ils.Scope().SetName(receiverName)
```

Comment:
You could consider adding a `scope` field to the `transaction` struct, and initializing the scope in `newTransaction`. Then you could copy the `scope` here instead of recreating it each time. For example:

```
type transaction struct {
  ...
  scope          pcommon.InstrumentationScope
  ...
```

```
func newTransaction() {
  ...
  scope := pcommon.NewInstrumentationScope()
  scope.SetName(scopeName)
  scope.SetVersion(buildInfo.Version)
  ...
}
```

then here you can:

```
ils := rms.ScopeMetrics().AppendEmpty()
t.scope.CopyTo(ils.Scope())
```

Reply:
Currrent implementation 
```
BenchmarkReceiverVersionAndNameAreAttached-10    	  302818	      4001 ns/op
```
Proposed one
```
BenchmarkReceiverVersionAndNameAreAttached-10    	  304873	      4082 ns/op
```

Reply:
It seems negligible at best. I'm happy with the current implementation.